src/Microsoft.ML.Data/DataLoadSave/Text/TextSaver.cs

```diff
@@ -116,28 +116,28 @@ protected ValueWriterBase(PrimitiveType type, int source, char sep)
                     Conv = Conversions.Instance.GetStringConversion<T>(type);
 
                 var d = default(T);
-                Conv(ref d, ref Sb);
+                Conv(in d, ref Sb);
                 Default = Sb.ToString();
             }
 
-            protected void MapText(ref ReadOnlyMemory<char> src, ref StringBuilder sb)
+            protected void MapText(in ReadOnlyMemory<char> src, ref StringBuilder sb)
             {
                 TextSaverUtils.MapText(src.Span, ref sb, Sep);
             }
 
-            protected void MapTimeSpan(ref TimeSpan src, ref StringBuilder sb)
+            protected void MapTimeSpan(in TimeSpan src, ref StringBuilder sb)
```

Comment:
Just a note: some of these don't seem worthwhile to pass by `in`, since they are often smaller than a register in size.

Reply:
As we see though we benefit from having a uniform signature no matter what the input type is, even if not all types will benefit, since we can then use the same high level code for all types.

Reply:
These have to be changed to `in` because these methods are used as `ValueMapper` delegates.

https://github.com/dotnet/machinelearning/blob/a8bc26820c2b33f2010b39d5dff4f9d69f9922b2/src/Microsoft.ML.Data/DataLoadSave/Text/TextSaver.cs#L100-L114